python/ray/serve/autoscaling_policy.py

```diff
@@ -5,91 +5,17 @@
-            been time-averaged over the desired lookback window.
-        override_min_replicas: Overrides min_replicas from the config
-            when calculating the final number of replicas.
-        override_max_replicas: Overrides max_replicas from the config
-            when calculating the final number of replicas.
-
-    Returns:
-        desired_num_replicas: The desired number of replicas to scale to, based
-            on the input metrics and the current number of replicas.
-
-    """
-    current_num_replicas = len(current_num_ongoing_requests)
-    if current_num_replicas == 0:
-        raise ValueError("Number of replicas cannot be zero")
-
-    # The number of ongoing requests per replica, averaged over all replicas.
-    num_ongoing_requests_per_replica: float = sum(current_num_ongoing_requests) / len(
-        current_num_ongoing_requests
-    )
-
-    # Example: if error_ratio == 2.0, we have two times too many ongoing
-    # requests per replica, so we desire twice as many replicas.
-    error_ratio: float = (
-        num_ongoing_requests_per_replica
-        / autoscaling_config.target_num_ongoing_requests_per_replica
-    )
-
-    # If error ratio >= 1, then the number of ongoing requests per
-    # replica exceeds the target and we will make an upscale decision,
-    # so we apply the upscale smoothing factor. Otherwise, the number of
-    # ongoing requests per replica is lower than the target and we will
-    # make a downscale decision, so we apply the downscale smoothing
-    # factor.
-    if error_ratio >= 1:
-        smoothing_factor = autoscaling_config.get_upscale_smoothing_factor()
-    else:
-        smoothing_factor = autoscaling_config.get_downscale_smoothing_factor()
-
-    # Multiply the distance to 1 by the smoothing ("gain") factor (default=1).
-    smoothed_error_ratio = 1 + ((error_ratio - 1) * smoothing_factor)
-    desired_num_replicas = math.ceil(current_num_replicas * smoothed_error_ratio)
-
-    # If error_ratio = 0, meaning there is no more traffic, and desired
-    # num replicas is stuck at a positive number due to the math.ceil
-    # above, decrease desired_num_replicas by one so that the deployment
-    # can eventually scale to 0.
-    if (
-        error_ratio == 0
-        and desired_num_replicas == current_num_replicas
-        and desired_num_replicas >= 1
-    ):
-        desired_num_replicas -= 1
-
-    min_replicas = autoscaling_config.min_replicas
-    max_replicas = autoscaling_config.max_replicas
-    if override_min_replicas is not None:
-        min_replicas = override_min_replicas
-    if override_max_replicas is not None:
-        max_replicas = override_max_replicas
-
-    # Ensure scaled_min_replicas <= desired_num_replicas <= scaled_max_replicas.
-    desired_num_replicas = max(min_replicas, min(max_replicas, desired_num_replicas))
-
-    return desired_num_replicas
-
-
+@PublicAPI(stability="stable")
```

Comment:
Should we mark these as `stable` now? `alpha`/`beta` seems safer until we get more users using custom policies.

Reply:
Yea, I was debating whether this should be stable or alpha/ beta and chose to mark it stable due to it's the current default, will mark it as beta instead

Reply:
These should all be marked `alpha` for now